src/common/rt64_apple.h

```diff
@@ -13,6 +13,28 @@ struct CocoaWindowAttributes {
 
 const char* GetHomeDirectory();
 
-void GetWindowAttributes(void* window, CocoaWindowAttributes *attributes);
-int GetWindowRefreshRate(void* window);
-void WindowToggleFullscreen(void* window);
+class CocoaWindow {
+private:
+    void* windowHandle;
+    std::atomic<int> cachedX;
+    std::atomic<int> cachedY;
+    std::atomic<int> cachedWidth;
+    std::atomic<int> cachedHeight;
+    std::atomic<int> cachedRefreshRate;
```

Comment:
The usage of atomics leaves one small possibility that you can get intermediate updates where only some of the attributes were updated partially. It's a very small possibility, but it's worth pointing out what downside it has versus just using the mutex around the four attributes.